jdaviz/configs/cubeviz/helper.py

```diff
@@ -126,3 +130,60 @@ class CubeViz(Cubeviz):
 
 def layer_is_cube_image_data(layer):
     return isinstance(layer, BaseData) and layer.ndim in (2, 3)
+
+
+# TODO: We can remove this when specutils supports it, i.e.,
+#       https://github.com/astropy/specutils/issues/592 and
+#       https://github.com/astropy/specutils/pull/1009
+# NOTE: Cannot use custom_write decorator from specutils because
+#       that involves asking user to manually put something in
+#       their ~/.specutils directory.
+
+def jdaviz_cube_fitswriter(spectrum, file_name, **kwargs):
+    """This is a custom writer for Spectrum1D data cube.
+    This writer is specifically targetting data cube
+    generated from Cubeviz plugins (e.g., cube fitting)
+    with FITS WCS. It writes out data in the following format
+    (with MASK only exist when applicable)::
+
+        No.    Name      Ver    Type
+          0  PRIMARY       1 PrimaryHDU
+          1  SCI           1 ImageHDU (float32)
+          2  MASK          1 ImageHDU (uint16)
+
+    Examples
+    --------
+    To write out a Spectrum1D cube using this writer:
+
+    >>> spec.write("my_output.fits", format="jdaviz-cube-writer", overwrite=True)  # doctest: +SKIP
```

Comment:
should there be an example on how to re-load the resulting fits file back into a Spectrum1D or is that just considered obvious?  Does that require providing the format again?

Reply:
I think you can just use vanilla `Spectrum1D.read`. Existing Cubeviz parser can read it back.